File: BackEnd/LRU.cs
using System.Collections.Generic;
using System.Linq;

namespace MeiSolver.BackEnd {
    class LRU {
        private List<int> values;
        private int lastIndex;

        public LRU(int elements) {
            //values = 3,2,1,0, so the cache will use 0 as the first
            values = Enumerable.Range(0, elements).Reverse().ToList();
            lastIndex = elements - 1;
        }

        public int getLeastRecentlyUsed() {
            return values[lastIndex];
        }

        public void pushToTop(int index) {
            values.Remove(index);
            values.Insert(0, index);
        }
    }
}

File: src/org/architecture_helper/backend/LRU.java
package org.architecture_helper.backend;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

class LRU {
    private List<Integer> values;
    private int lastIndex;

    public LRU(int elements) {
        //values = 3,2,1,0, so the cache will use 0 as the first
        values = IntStream.range(0, elements).boxed().collect(Collectors.toList());
        Collections.reverse(values);
        lastIndex = elements - 1;
    }

    public int getLeastRecentlyUsed() {
        return values.get(lastIndex);
    }

    public void pushToTop(int index) {
        values.remove(index);
        values.add(0, index);
    }
}